doc/protocol.txt

```diff
@@ -400,6 +444,366 @@ VALUE <key> <flags> <bytes> [<cas unique>]\r\n
 
 - <data block> is the data for this item.
 
+Meta Debug
+----------
+
+The meta debug command is a human readable dump of all available internal
+metadata of an item, minus the value.
+
+me <key>\r\n
+
+- <key> means one key string.
+
+The response looks like:
+
+ME <key> <k>=<v>*\r\nEN\r\n
+
+Each of the keys and values are the internal data for the item.
+
+exp   = expiration time
+la    = last access time
+cas   = CAS ID
+fetch = whether an item has been fetched before
+cls   = slab class id
+size  = total size in bytes
+
+Others may be added.
+
+Meta Get
+--------
+
+The meta get command is the generic command for retrieving key data from
+memcached. Based on the flags supplied, it can replace all of the commands:
```

Comment:
Are we planning to support multiple keys with meta commands in the future?

Reply:
I don't think so; based on dealing with ascii-multiget (which only works with get!) it makes the network code horrible. In my next batch of work re-doing the network code it should be possible to batch syscalls for any stack of pipelined commands. So "me key1\r\nme key2\r\n" and then you'd get one syscall for all the responses, except it'll also work with set/delete/etc.

For now it'll be equivalent to the binary protocol which batches nothing ever :)